renode-linux-runner-docker/payload/run-in-renode.py

```diff
@@ -19,7 +19,35 @@
 from subprocess import run, DEVNULL, CalledProcessError
 from sys import stdout as sys_stdout, exit as sys_exit, argv as sys_argv
 from time import sleep
+from re import sub as re_sub, compile as re_compile
 
+CR = r'\r'
+
+class FilteredStdout(object):
+    """
+    Stdout wrapper which replaces found pattern with 'replace' string.
+    It is used to remove \r characters from telnet output. GitHub workflow logs
+    interpret this sequence as newline.
+    """
+    def __init__(self, stream, pattern: str, replace: str):
+        """
+        Parameters
+        ----------
+        stream : stdout
+        pattern : regex pattern to match and replace with 'replace'
+        replace : string to replace found patterns
+        """
+        self.stream = stream
+        self.pattern = re_compile(pattern)
+        self.replace = replace
+    
```

Comment:
Remove trailing whitespace. I think we'll want to set up some formatter thing for this repo (not in the scope of this pull though)